src/classes/ADDR_OtherAcc_TEST.cls

```diff
@@ -0,0 +1,204 @@
+ 
+        Integer cAcc = 3;
+        list<Account> listAcc = UTIL_UnitTestData_TEST.CreateMultipleTestAccounts(cAcc, UTIL_Describe.getBizAccRecTypeID());
+        for (Account acc : listAcc) {
+            acc.BillingStreet = '123 45th';
+            acc.BillingCity = 'Seattle';
+        }
+        Test.startTest();
+        insert listAcc;
+        Test.stopTest();
+
+        // verify results
+        map<Id, Account> mapAccIdAcc = new map<Id, Account>([select Id, Name, BillingStreet, BillingCity, BillingState, 
+        BillingPostalCode, BillingCountry, BillingLatitude, BillingLongitude, is_Address_Override__c, Current_Address__c from Account]);
+        system.assertEquals(cAcc, mapAccIdAcc.size());
+        
+        list<Address__c> listAddr = [select Id, Default_Address__c, MailingStreet__c, MailingCity__c, Parent_Account__c, 
+        Latest_Start_Date__c, Latest_End_Date__c from Address__c];
+        system.assertEquals(cAcc, listAddr.size());
+        
+        for (Address__c addr : listAddr) {
+            Account acc = mapAccIdAcc.get(addr.Parent_Account__c);
+            system.assertEquals(acc.BillingStreet, addr.MailingStreet__c);
+            system.assertEquals(acc.BillingCity, addr.MailingCity__c);
+            system.assertEquals(true, addr.Default_Address__c);
+        }
+    }
+    
+    /*********************************************************************************************************
+    @description Update N organizational accounts with billing addresses 
+    verify:
+        N addresses created
+        account addresses updated
+    **********************************************************************************************************/            
+    @isTest
+    public static void updateRecordsWithAddrInfo() {         
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Accounts_Addresses_Enabled__c = 
+                                                                            UTIL_Describe.getBizAccRecTypeID() + ';'));
+ 
+        // create accounts without addresses
+        Integer cAcc = 3;
+        list<Account> listAcc = UTIL_UnitTestData_TEST.CreateMultipleTestAccounts(cAcc, UTIL_Describe.getBizAccRecTypeID());
+        insert listAcc;
+
+        // update the accounts' address
+        for (Account acc : listAcc) {
+            acc.BillingStreet = '123 45th';
+            acc.BillingCity = 'Seattle';
+        }
+        Test.startTest();
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update listAcc;
+        Test.stopTest();
+
+        // verify results
+        listAcc = [select Id, Name, BillingStreet, BillingCity, BillingState, BillingPostalCode, BillingCountry, 
+                   Current_Address__c from Account order by Account.Name];
+        system.assertEquals(cAcc, listAcc.size());
+        
+        list<Address__c> listAddr = [select Id, Default_Address__c, MailingStreet__c, MailingCity__c, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.